REPORT.md

```diff
@@ -0,0 +1,145 @@
+| 2  	| litiengine/src/de/gurkenlabs/litiengine/environment/Environment.java           	| add                   	| 18   	| 17   	|
+| 3  	| litiengine/src/de/gurkenlabs/litiengine/environment/Environment.java           	| remove                	| 18   	|      	|
+| 4  	| litiengine/src/de/gurkenlabs/litiengine/environment/MapObjectSerializer.java   	| getPropertyValue      	| 18   	| 18   	|
+| 5  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/MapUtilities.java  	| getTile               	| 18   	| 17   	|
+| 6  	| litiengine/src/de/gurkenlabs/litiengine/environment/CustomMapObjectLoader.java 	| invoke                	| 17   	|      	|
+| 7  	| litiengine/src/de/gurkenlabs/litiengine/graphics/RenderComponent.java          	| render                	| 17   	|      	|
+| 8  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/xml/Layer.java     	| afterUnmarshal        	| 16   	| 15   	|
+| 9  	| litiengine/src/de/gurkenlabs/litiengine/environment/tilemap/MapUtilities.java  	| assessHexStaggering   	| 16   	| 15   	|
+| 10 	| litiengine/src/de/gurkenlabs/litiengine/util/geom/GeometricUtilities.java      	| getIntersectionPoints 	| 16   	| 16   	|
+
+###### a. Did all tools/methods get the same result?
+The calculations we performed resulted in very similar results. However, something that the calculations had in common, compared to the results from running lizard, was that they were slightly lower in general (see function #2, #5, #8 and #9). 
+
+###### b. Are the results clear?
+The functions were relative straightforward which made it easy to graph. As such, it was somewhat clear how lizard calculated the CNN, given that the hand calculated CCN gave a similar result to lizard's CCN.   
+
+##### 2. Are the functions just complex, or also long?
+The length of the functions is varying, from around 10 lines to 75. In other words, not all complex functions in this project are very long.
+
+##### 3. What is the purpose of the functions?
+The purpose of the functions also vary. Some are regarding the environmental setup of the engine, thus they are complex. Others are functions related to utilities, e.g. finding a point that intersects two lines. 
+
+##### 4. Are exceptions taken into account in the given measurements?
+Some of the methods examined, e.g. `storeProperties`, handles exceptions. This increased the CC count by one, since it increases the possible paths through the function by one. 
+
+##### 5. Is the documentation clear w.r.t. all the possible outcomes?
+The documentation regarding the possible paths is varying. The method `getIntersectionPoint` does not explain the possible paths, for example if a point does not exist on the lines it is not defined what the result will be. In general the documentation of the different paths possible and their corresponding results is lacking. Some of the methods examined do not have a method comment, e.g. `afterUnmarshal`.
```

Comment:
`getIntersectionPoint` -> `getIntersectionPoints`.

` for example if a point does not exist on the lines it is not defined what the result will be` ->
` for example if an intersection point does not exist, it is not defined what the result will be`